src/util/resolveName.ts

```diff
@@ -0,0 +1,13 @@
+import { EdgeCurrencyWallet } from 'edge-core-js'
+
+import { ResolutionError } from '../types/ResolutionError'
+
+export const resolveName = async (wallet: EdgeCurrencyWallet, name: string): Promise<string> => {
+  if (!('resolveName' in wallet.otherMethods)) {
+    throw new ResolutionError('UnsupportedDomain', { domain: name })
+  }
```

Comment:
`if (typeof wallet.otherMethods.resolveName !== 'function')` might be more picky.